Первое решение в файле g0.py использует рекурсию, из-за чего на 23 тесте происходила ошибка исполнения.
Я переписал с помощью стеков (в файле g.py), но получил неравильный ответ на 19 тесте.
Чтобы найти ошибку в коде, я написал тест (test_g.py), где сравнивал ответы g и g0.
Нашел пример поменьше и понял, что g0 записывает в tout один элемент несколько раз.
Для исправления этой ошибки, я добавил второе условие в dfs1.
